### PR TITLE
Create Meetup Photo

### PR DESCRIPTION
**Pull request by schmidtwmark**
Put the camera in the app screen. For some unknown reason the picture it takes is rotated.

EDIT: it fucking works. I used a different tutorial this time and it made things so much easier. Still need to add a retake button to the edit page, but I'll save that for when the edit page gets redone.

### CODE REVIEW

File: app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java [hunk 1]
```diff
@@ -1,21 +1,84 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CreateMeetupPhotoActivity extends CreateMeetupActivity {
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java]: This is a lot of code that doesn't have to do with the activity itself, and makes navigating this class very difficult.  I'd recommend refactoring almost every new function here to one or more classes somewhere else.

File: app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java [hunk 2]
```diff
@@ -1,53 +1,256 @@
...
-    static final int REQUEST_IMAGE_CAPTURE = 1;
+    CameraManager cameraManager;
+    int cameraFacing;
 
-    private ImageView mImageView;
-    private Button mNextButton;
+    TextureView.SurfaceTextureListener surfaceTextureListener;
+    Size previewSize;
+    String cameraId;
+    private HandlerThread backgroundThread;
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java]: How about refactoring these members so they start with m?

File: app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java [hunk 3]
```diff
@@ -1,53 +1,256 @@
...
+        textureView = findViewById(R.id.camera_preview);
+        surfaceTextureListener = new TextureView.SurfaceTextureListener() {
+            @Override
+            public void onSurfaceTextureAvailable(SurfaceTexture surfaceTexture, int width, int height) {
+                setUpCamera();
+                openCamera();
+            }
+
+            @Override
+            public void onSurfaceTextureSizeChanged(SurfaceTexture surfaceTexture, int width, int height) {
```
Comment (divy4) [inline app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java]: Is it necessary to write these functions?

Comment (schmidtwmark) [reply app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java]: Apparently they are required

Comment (divy4) [reply app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupPhotoActivity.java]: Huh, alright then

**Pull request merged**